tests/utils/test_configdict.py

```diff
@@ -0,0 +1,129 @@
+import pytest
+
+from impactlab_tools.utils.configdict import ConfigDict, gather_configtree
+
+
+@pytest.fixture
+def simple_nested_tree():
+    return {'a': 1, 'b': {'a': 2}, 'c': 3, 'd-4': 4, 'e_5': 5, 'F': 6}
+
+
+def test_configdict_climbs_tree(simple_nested_tree):
+    conf = ConfigDict(simple_nested_tree)
+    assert conf['c'] == 3
+
+
+def test_configdict_prefer_child(simple_nested_tree):
+    conf = gather_configtree(simple_nested_tree)['b']
+    assert conf['a'] == 2
+
+
+def test_configdict_selfreference(simple_nested_tree):
+    conf = gather_configtree(simple_nested_tree)
+    conf['b'] = conf
+    assert conf['b'] == conf
+
+
+def test_configdict_caseinsensitive_keys(simple_nested_tree):
+    conf = gather_configtree(simple_nested_tree)
+    assert conf['A'] == 1
+    assert conf['f'] == 6
+
+    conf['Z'] = 'foobar'
+    assert conf['z'] == 'foobar'
+
+
+def test_configdict_normalize(simple_nested_tree):
+    """Normalize underscores and hyphens in keys
+    """
+    conf = ConfigDict(simple_nested_tree)
+    assert conf['d_4'] == 4
+    assert conf['e-5'] == 5
+
+    conf['y-7'] = 'foobar'
+    assert conf['y_7'] == 'foobar'
+
+    conf['z-9'] = 'foobar'
+    assert conf['z_9'] == 'foobar'
+
+
+def test_configdict_throws_keyerror(simple_nested_tree):
+    conf = ConfigDict(simple_nested_tree)
+    with pytest.raises(KeyError):
+        conf['foobar']
+
+
+def test_configdict_merge(simple_nested_tree):
+    conf1 = ConfigDict(simple_nested_tree)
+    conf2 = ConfigDict(simple_nested_tree)
+    conf2['a'] = 9
+
+    conf_merge = conf1.merge(conf2)
+
+    assert conf_merge.data == conf2.data
+    assert conf_merge.parent == conf1.parent
```

Comment:
Can you just make one of these dictionaries something like `{'foo': 0}`, and then show that you can access both entries in `simple_nested_tree` and access `foo`?

Reply:
Done in new commit.